generated/Order.h

```diff
@@ -0,0 +1,112 @@
+#ifndef ORG_OPENAPITOOLS_MODEL_Order_H_
+#define ORG_OPENAPITOOLS_MODEL_Order_H_
+
+#include "serialization.h"
+
+#include <optional>
+
+#include "DateTime.h"
+#include <string>
+
+namespace org {
+namespace openapitools {
+namespace model {
+
+/**
+ * \brief An order for a pets from the pet store
+ */
+struct Order {
+    static const char *dtoName() { return "swagger::dto::Order"; }
+
+    // enums
+    /**
+     * \brief Order Status
+     */
+    enum StatusEnum {
+        PLACED,
+        APPROVED,
+        DELIVERED,
+    };
+
+
+    // optional fields
+    std::optional<int64_t> id;
+    std::optional<int64_t> petId;
+    std::optional<int32_t> quantity;
+    std::optional<DateTime> shipDate;
+    std::optional<StatusEnum> status;
+    std::optional<bool> complete;
+};
+
+void to_json(nlohmann::json& j, const Order& obj) {
+    j = nlohmann::json{
+        {"id", obj.id},
+        {"petId", obj.petId},
+        {"quantity", obj.quantity},
+        {"shipDate", obj.shipDate},
+        {"status", obj.status},
+        {"complete", obj.complete},
+    };
+}
+
+void from_json(const nlohmann::json& j, Order& obj) {
+    detail::to_optional(j, "id", obj.id);
+    detail::to_optional(j, "petId", obj.petId);
+    detail::to_optional(j, "quantity", obj.quantity);
+    detail::to_optional(j, "shipDate", obj.shipDate);
+    detail::to_optional(j, "status", obj.status);
+    detail::to_optional(j, "complete", obj.complete);
+}
+
+}  // namespace org
+}  // namespace openapitools
+}  // namespace model
+
+namespace nlohmann {
+
+template<>
+struct adl_serializer<org::openapitools::model::Order::StatusEnum> {
+  private:
+    using Order = org::openapitools::model::Order;
+
+  public:
+    static void to_json(json& j, const Order::StatusEnum& value) {
+        switch (value) {
+        case Order::StatusEnum::PLACED:
+            j = "PLACED";
+            return;
+        case Order::StatusEnum::APPROVED:
+            j = "APPROVED";
+            return;
+        case Order::StatusEnum::DELIVERED:
+            j = "DELIVERED";
+            return;
+        }
+        UNREACHABLE_CODE
+    }
+
+    static void from_json(const json& j, Order::StatusEnum& value) {
+        // TODO: there are some iffy assumptions about casing here
+        const std::string s = j.get<std::string>();
+        std::string all_caps;
+        std::transform(s.cbegin(), s.cend(), std::back_inserter(all_caps), ::toupper);
```

Comment:
I believe it would be totally fair to assume that the backend always sends us "all caps" and we don't need to do any fancy case conversion here. Do check with them, though.

Reply:
I agree. Come to think of it, it's also not a good practice to try to "fix" the user input when sanitizing.